cupy/manipulation/add_remove.py

```diff
@@ -12,8 +13,47 @@
 
 # TODO(okuta): Implement resize
 
+_first_nonzero_krnl = core.ReductionKernel(
+    'T data, int64 len',
+    'int64 y',
+    'data == T(0) ? len : _j',
+    'min(a, b)',
+    'y = a',
+    'len',
+    'first_nonzero'
+)
 
-# TODO(okuta): Implement trim_zeros
+
+def trim_zeros(filt, trim='fb'):
+    """Trim the leading and/or trailing zeros from a 1-D array or sequence.
+
+    Returns the trimmed array
+
+    Args:
+        filt(cupy.ndarray): Input array
+        trim(str, optional):
+            'fb' default option trims the array from both sides.
+            'f' option trim zeros from front.
+            'b' option trim zeros from back.
+
+    Returns:
+        cupy.ndarray: trimmed input
+
+    .. seealso:: :func:`numpy.trim_zeros`
+
+    """
+    if filt.ndim > 1:
+        raise ValueError('Multi-dimensional trim is not supported')
+    if not filt.ndim:
+        raise TypeError('0-d array cannot be trimmed')
+    start = 0
+    end = filt.size
+    trim = trim.upper()
+    if 'F' in trim:
+        start = _first_nonzero_krnl(filt, filt.size)
+    if 'B' in trim:
+        end = filt.size - _first_nonzero_krnl(filt[::-1], filt.size)
```

Comment:
Please convert 0-dim cupy.ndarray to scalar before 1-size GPU operation for the performance?
```suggestion
        end = filt.size - _first_nonzero_krnl(filt[::-1], filt.size).item()
```
`start = _first_nonzero_krnl(filt, filt.size).item()` is also preferred for the consistency.